cluster/ci/monitoring/mixins/prometheus/ghproxy.libsonnet

```diff
@@ -0,0 +1,45 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'ghproxy',
+        rules: [
+          {
+            alert: 'ghproxy-status-code-abnormal-%s' % std.strReplace(code, '..', 'XX'),
+            // excluding 404 because otherwise
+            // "this is going to spam us just about 24/7 with the OWNERS thing"
+            expr: |||
+              sum(rate(github_request_duration_count{status=~"%s", status!="404"}[5m])) > 0
+            ||| % code,
+            'for': '1m',
+            labels: {
+              severity: 'slack',
+            },
+            annotations: {
+              message: 'ghproxy has a increasing number of status code %s.' % std.strReplace(code, '..', 'XX'),
```

Comment:
How often does this fire today? I worry that a non-zero positive rate for 1min will happen every time we see _any_

Reply:
https://ibb.co/h9TPFfM

6 times in the last 24 hours.

If my calculation is correct, based on my observation for the data today:

```
for (m): No. alerts
1: 6
15: 2
30: 1
60: 0
```

Maybe go for `15m`?

Reply:
Depends -- which response codes are we hitting? Let's alert on every instance where an admin should actually do something.

Reply:
[status](https://golang.org/src/net/http/status.go) 4..|5.. in the last 24 hours

```
403: 8 StatusForbidden
404: 1800 StatusNotFound (ignored for now)
405: 24 StatusMethodNotAllowed
422: 4 StatusUnprocessableEntity
500: 2 StatusInternalServerError

3..: all 304 ... it is OK. (ignored for now)
```
It would be hard (not worth) digging into each of them, maybe even should not.

Can I think in another way?
Inogring 304 and 404, we have a good day. I guess i can claim that.
rate>0 might be too strict as you pointed out.

maybe like our (un)success percentage ...
```
sum(github_request_duration_count{status=~"n..", status!="404"})/sum(github_request_duration_count{status!="404"}) for n in {4..5}
```
Say tolerance is 0.05 (we can learn by time)? Totay<0.04 all day long.

Maybe makes more sense this way.

Reply:
I like that -- alert when that unsuccessful percentage goes above the reasonable level